### PR TITLE
[4.0] whitelist strings

### PR DESCRIPTION
**Pull request by brianteeman**
~Updates the strings to use Allowed and Forbidden instead of whitelist and blacklist. This still leaves the use of whitelist in com_csp as I can't actually see a list at all. @zero-24 please clarify~

Uses the terms allowed and forbidden instead of whitelist and blacklist

Continuation of #29703

Previous comment

>I just spent some time looking at this and it will never be a simple search and replace. After thinking about it for a while I think in most cases it will be enough to just changed to allowed and forbidden but there will be some occasions where for linguistic reasons it will need to be allowed list and forbidden list

>The second issue is should this be everywhere or just in the visible strings. Everywhere means changing comments and variable names and to me the time to do that has now passed. But some might argue that if you still have a variable called $blacklist then you're not really desensitizing the language.

If @wilsonge and @bembelimen can make a decision on that I can easily do the work

### CODE REVIEW

File: administrator/language/en-GB/com_media.ini [hunk 1]
```diff
@@ -33,7 +33,7 @@ COM_MEDIA_FIELD_CHECK_MIME_DESC="Use MIME Magic or Fileinfo to attempt to verify
 COM_MEDIA_FIELD_CHECK_MIME_LABEL="Check MIME Types"
 COM_MEDIA_FIELD_IGNORED_EXTENSIONS_DESC="Ignored file extensions for MIME type checking and restricted uploads."
 COM_MEDIA_FIELD_IGNORED_EXTENSIONS_LABEL="Ignored Extensions"
-COM_MEDIA_FIELD_ILLEGAL_MIME_TYPES_DESC="A comma separated list of illegal MIME types to upload (blacklist)."
+COM_MEDIA_FIELD_ILLEGAL_MIME_TYPES_DESC="A comma separated list of forbidden MIME types to upload."
 COM_MEDIA_FIELD_ILLEGAL_MIME_TYPES_LABEL="Illegal MIME Types"
```
Comment (Quy) [inline administrator/language/en-GB/com_media.ini]:
```suggestion
COM_MEDIA_FIELD_ILLEGAL_MIME_TYPES_LABEL="Forbidden MIME Types"
```

**Comment by zero-24** [comment]
The usage of whitelist in com_csp can be changed to allowlist as well without any problems.

**Comment by brianteeman** [comment]
But where is this list?

**Comment by zero-24** [comment]
The collected and approved urls are put together as 'allowlist' for example or in this case Allowed domains to "use" whatever directive we are talking about.

**Comment by wilsonge** [comment]
For the code. It's going to be a bit situational. For things like https://github.com/joomla/joomla-cms/pull/29703/files#diff-108363cbc354ffab84b1fd1e834d0406f74132c0dd946ec7f8445e3c4ad67e1bR136 which was soley a variable used in a function (i.e. not accessible to anyone) then it's fine to change obviously. 

Some of the other changes like https://github.com/joomla/joomla-cms/pull/29703/files#diff-e6d69a506c2984b154921451397a94d38a0513f0f40da9a531cc311a40daf624R145 which changed the term saved to db - less sure. As we're still going to always read both formats out not really sure it's a major issue (and personally it's one of those things where I wouldn't even deprecate for bs5 - just read them both forever - because assets are rarely going to change and I think it's asking too much)

I think the major thing in the old PR I wouldn't maybe feel comfortable changing anymore would be the ListModel stuff - because that's very widely extended by 3rd parties.

**Comment by bembelimen** [comment]
Thank you for your PR! In my opinion we should do all or should not do it at all. Otherwise it is just a "sham". That's what I tried at least with the PR. I think I found all the appearance and made it B/C, so you could just look up the lines and fix them. My problem was just to be not native english...

**Comment by brianteeman** [comment]
@bembelimen I will take a look this weekend. The key issue is that its not a simple search replace so each use has to be checked.

**Comment by adj9** [comment]
I have not tested this item.<br /><br />But where are the terms to be verified?<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/32030">issues.joomla.org/tracker/joomla-cms/32030</a>.</sub>

**Comment by brianteeman** [comment]
@bembelimen I _think_ I have got all the code changes you suggested plus one additional one. Please can you check.

File: administrator/components/com_users/src/Model/UsersModel.php [hunk 2]
```diff
@@ -28,13 +28,22 @@
...
 	 */
 	protected $filterBlacklist = array('groups', 'excluded');
 
+	/**
+	 * A list of filter variables to not merge into the model's state
+	 *
+	 * @var    array
+	 * @since  __DEPLOY_VERSION__
+	 */
+	protected $filterDisallowList = array('groups', 'excluded');
```
Comment (Quy) [inline administrator/components/com_users/src/Model/UsersModel.php]: `$filterForbiddenList`?

**Comment by bembelimen** [comment]
Didn't found anything, not sure if this b/c thing in the UsersModel is needed, because it's merged anyways, but otherways great job, thanks 👍

**Comment by bembelimen** [comment]
I think you can remove these lines: https://github.com/joomla/joomla-cms/pull/32030/files#diff-d3947ac81429b01036f6815ca8505d5d6f91ac349787cf808a42d23b7081cf9dR30-R37

**Comment by brianteeman** [comment]
@bembelimen not sure as I'm not sure about the deprecation at such a late time. Will leave that decision to @wilsonge

**Comment by ceford** [comment]
I have tested this item :white_check_mark: successfully on 25686ef6423816af8a3be8ade7b9fb7a0f44b554<br /><br />I code searched for BLACKLIST, BLACK_LIST, blacklist and Blacklist (and white versions) before and after applying PR. The only ones unaffected were in libraries/vendor and media/vendor. Except for Blacklist in libraries/src/MVC/Model/ListModel.php - was that covered in the discussion?<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/32030">issues.joomla.org/tracker/joomla-cms/32030</a>.</sub>

**Comment by wilsonge** [comment]
For the component specific implementation I think it's fine to remove the old properties. In list model you're right to keep B/C

**Comment by ghost** [comment]
I have tested this item :white_check_mark: successfully on 25686ef6423816af8a3be8ade7b9fb7a0f44b554<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/32030">issues.joomla.org/tracker/joomla-cms/32030</a>.</sub>

**Comment by brianteeman** [comment]
@wilsonge I'm not sure I understand what you want to do here.

**Comment by brianteeman** [comment]
I guess its not that important after all

**Comment by wilsonge** [comment]
Thanks!

**Comment by brianteeman** [comment]
Now I see your commit I understand your comment. I didnt before - sorry

**Comment by wilsonge** [comment]
Don't worry about it. Sorry for not replying promptly!

**Pull request merged**